pkg/apiserver/handler.go

```diff
@@ -110,6 +115,10 @@ func CreateJsonHandler(handler HandlerWithInput) gin.HandlerFunc {
 	return handleWithInput(handler, binding.JSON, defaultErrorHandler)
 }
 
+func CreateJsonAndQueryHandler(handler HandlerWithMultipleBindings) gin.HandlerFunc {
```

Comment:
since you are just wrapping `handleWithChainedInputs`, i would suggest to name the function `CreateChainedHandler`